### PR TITLE
ft: extend retrieveData parameters

### PR DESCRIPTION
**Pull request by electrachong**
Dependency of https://github.com/scality/Mystique/pull/18

### CODE REVIEW

**Comment by electrachong** [comment]
@ironman-machine
SCALITY_S3_BRANCH=rf/s3header-validators
try

**Comment by electrachong** [comment]
@ironman-machine
SCALITY_S3_BRANCH=rf/extract-header-validators
try

**Comment by electrachong** [comment]
@ironman-machine
SCALITY_S3_BRANCH=rf/retrieve-data-params
try

**Comment by electrachong** [comment]
end-to-end retried after pushing a fix: http://ci.ironmann.io/gh/scality/Integration/13908

File: lib/s3routes/routesUtils.js [hunk 1]
```diff
@@ -376,10 +376,13 @@ const routesUtils = {
      * @param {array | undefined} range - range in format of [start, end]
      * if range header contained in request or undefined if not
      * @param {object} log - Werelogs logger
+     * @param {object} [extraParams] - object containing extra params
```
Comment (LaurenSpiegel) [inline lib/s3routes/routesUtils.js]: please remind me what this is needed for. can we include something in the comment that is generic but also gives an idea of what is going?

Comment (electrachong) [inline lib/s3routes/routesUtils.js]: it's needed to supply any additional information, like the bucket name or formatted range, if needed for the backend to stream data.

**Comment by electrachong** [comment]
@ironman-machine
SCALITY_S3_BRANCH=rf/retrieve-data-params
try

**Comment by electrachong** [comment]
end-to-end retried after broken ci repaired: http://ci.ironmann.io/gh/scality/Integration/14028

**Comment by electrachong** [comment]
Changing the way I pass params from Mystique so we don't need to extend them in dataRetrievalFn; no need for the pr so closing.

**Pull request closed without merging**